.github/workflows/e2e_ci.yml

```diff
@@ -96,7 +88,45 @@ jobs:
                 -DUSE_BUNDLED_LIBBPF=ON \
                 -DUSE_BUNDLED_GTEST=ON \
                 ..
-          make -j6
+          make -j6 libsinsp_e2e_tests
+          cd ..
+
+      - name: Cache build
+        uses: actions/cache/save@v3
+        if: always()
+        id: cache
+        with:
+          path: build
+          key: build-${{ matrix.arch }}-${{ github.run_id }}
+
+  test-e2e:
+    name: test-e2e-${{ matrix.arch }}-${{ matrix.driver.name }} 😇 (bundled_deps)
+    needs: [build-test-e2e]
+    runs-on: ${{ (matrix.arch == 'arm64' && 'actuated-arm64-8cpu-16gb') || 'ubuntu-22.04' }}
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+        driver: [ {name: kmod, option: -k}, {name: bpf, option: -b}, {name: modern-bpf, option: -m} ]
+      fail-fast: false
+    steps:
+      - name: Restore build skeleton
```

Comment:
Doesn't this mean that we always build tests, then cache the result, and then in `test-e2e` job we restore the just-cached build dir?
I think the proper way is to always try to restore-first (ie: keeping a single job) and if that fails, then build and cache for subsequent runs.

Reply:
This is just a first step: instead of building 6 times the same executable, this builds only once for arch. :)